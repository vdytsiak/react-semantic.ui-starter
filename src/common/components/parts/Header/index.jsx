/**
 * @flow
 */
import React from 'react'
import {Icon} from 'semantic-ui-react'
// import {FormattedMessage} from 'react-intl'
import {
	StyledHeader,
	HeaderInner,
	Navicon,
	PageTitle
	// HeaderButton
} from './style'
import {Spacer} from 'styles/base'
import Headroom from 'react-headroom'

type Props = {
	title: string,
	toggleSidebar: () => void,
	isLoggedIn: boolean,
	isMobile: boolean
}

const Header = ({title, toggleSidebar, isLoggedIn, isMobile}: Props) => {
	return (
		<Headroom>
			<StyledHeader>
				<HeaderInner>
					{isLoggedIn &&
						isMobile && (
							<Navicon onClick={toggleSidebar}>
								<Icon name="content" />
							</Navicon>
						)}
					<PageTitle>{title}</PageTitle>
					<Spacer />
				</HeaderInner>
			</StyledHeader>
		</Headroom>
	)
}

export default Header
