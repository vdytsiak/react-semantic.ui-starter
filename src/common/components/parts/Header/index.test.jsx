import React from 'react'
import Header from 'components/parts/Header'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'

const props = {
	title: 'Dashboard',
	toggleSidebar: () => ({}),
	isLoggedIn: true,
	isMobile: false
}

describe('Header component', () => {
	it('same snapshot at Dashboard on non-mobile screen', () => {
		const component = shallow(<Header {...props} />)
		expect(toJson(component)).toMatchSnapshot()
	})
})
