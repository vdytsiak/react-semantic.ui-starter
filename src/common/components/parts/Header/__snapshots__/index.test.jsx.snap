// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component same snapshot at Dashboard on non-mobile screen 1`] = `
<Headroom
  calcHeightOnResize={true}
  disable={false}
  disableInlineStyles={false}
  downTolerance={0}
  onPin={[Function]}
  onUnfix={[Function]}
  onUnpin={[Function]}
  parent={[Function]}
  pinStart={0}
  upTolerance={5}
  wrapperStyle={Object {}}
>
  <style__StyledHeader>
    <style__HeaderInner>
      <style__PageTitle>
        Dashboard
      </style__PageTitle>
      <base__Spacer />
    </style__HeaderInner>
  </style__StyledHeader>
</Headroom>
`;
